.circleci/create_circleci_config.py

```diff
@@ -401,6 +409,36 @@ def job_name(self):
     tests_to_run="tests/repo_utils",
 )
 
+doc_test_job = CircleCIJob(
+    "pr_documentation_tests",
+    additional_env={"TRANSFORMERS_VERBOSITY":"error", "DATASETS_VERBOSITY":"error", "SKIP_CUDA_DOCTEST": "1"},
+    install_steps=[
+        "sudo apt-get -y update && sudo apt-get install -y libsndfile1-dev espeak-ng time",
+        "pip install --upgrade pip",
+        "pip install -e .[dev]",
+        "pip install git+https://github.com/huggingface/accelerate",
+        "pip install --upgrade pytest pytest-sugar",
+        "find -name __pycache__ -delete",
+        "find . -name \*.pyc -delete",
+        {
+            "name": "Get files to test",
+            "command":
+                "git remote add upstream https://github.com/huggingface/transformers.git  && git fetch upstream \n"
+                "git diff --name-only --relative --diff-filter=AMR refs/remotes/upstream/main...HEAD | grep -E '\.(py|mdx)$' | grep -Ev '^\..*|/\.' | grep -Ev '__' > pr_documentation_tests.txt"
+        },
```

Comment:
~~I am wondering why we don't use the results given by `tests_fetcher.py` here.~~
Oh, because it only gives modified python files, but here you want also `.mdx` files.

Also, `refs/remotes/upstream/main` may not be the `main` in the local repo. on the runner. I am wondering why you want to compare the `remote main` against  local `HEAD` ?